com.unity.multiplayer.mlapi/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -114,6 +114,11 @@ internal void RemoveOwnership(NetworkObject networkObject)
                 writer.WriteUInt64Packed(networkObject.OwnerClientId);
 
                 NetworkManager.MessageSender.Send(NetworkConstants.CHANGE_OWNER, NetworkChannel.Internal, buffer);
+
+                foreach (var client in NetworkManager.ConnectedClients)
+                {
+                    NetworkManager.NetworkMetrics.TrackOwnershipChangeSent(client.Key, networkObject.NetworkObjectId, networkObject.name, (ulong)buffer.Length);
```

Comment:
I guess when I first saw this I figured you could just simply record the removal of ownership of OwnerClientId of NetworkObjectId vs. log removal of ownership as if all clients could have had it.  Or is this to simulate, say for bandwidth measurement, how the server notifies all listening clients of the change?  If the latter, it would seem more maintainable to put this kind of thing down inside NetworkManager.MessageSender.Send where it would just work for all different kinds of messages.

Reply:
Yeah this is initially where we wanted to put this kind of stuff, but unfortunately deeper in the stack we lose a lot of contextual information we want to keep along with these metrics. For example the name / id of the network object we're operating on.